kafka-connect-cassandra/src/main/scala/com/datamountaineer/streamreactor/connect/cassandra/sink/CassandraJsonWriter.scala

```diff
@@ -176,6 +188,51 @@ class CassandraJsonWriter(connection: CassandraConnection, settings: CassandraSi
   }
 
 
+  private def delete(records: Seq[SinkRecord]) = {
+    val executor = Executors.newFixedThreadPool(settings.threadPoolSize)
+
+    try {
+      val futures = records map { record =>
+        executor.submit {
+          try {
+            deleteCache match {
+              case Some(d) =>
+                val bindingFields = {
+                  if (record.keySchema().`type`().isPrimitive) {
```

Comment:
if json is sent this will result in NPE

https://github.com/datamountaineer/kafka-connect-common/blob/master/src/main/scala/com/datamountaineer/streamreactor/connect/converters/Transform.scala

how we handle all possible values